src/EnergyPlus/DXCoils.cc

```diff
@@ -8790,9 +8787,9 @@ void CalcHPWHDXCoil(EnergyPlusData &state,
 
     // determine inlet air temperature type for curve objects
     if (Coil.InletAirTemperatureType == WetBulbIndicator) {
-        InletAirTemp = HPWHInletWBTemp;
+        InletAirTemp = state.dataHVACGlobal->HPWHInletWBTemp;
```

Comment:
Opportunistically cleaning up some bad `auto &`'s as well.

Reply:
I found it interesting that DXCoils was removed from CppCheck when this was implemented. I wonder why?

cppcheck.yml

    -i EnergyPlus/DXCoils.cc \
    -i EnergyPlus/RefrigeratedCase.cc \
    -i EnergyPlus/SolarShading.cc \

Reply:
It's because the new version of cppcheck had some weird behavior and those files can take like 22 minutes to run through cppcheck now.  Julien created an issue on their issue tracker: https://trac.cppcheck.net/ticket/11587

When you want to clean up DXCoils, just turn that back on in the yaml file, it will take a while to run, but you'll get results, clean them up, and then turn it back off.